timelock-impl/src/main/java/com/palantir/atlasdb/timelock/auth/impl/PrivilegeBasedAuthRequirer.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.auth.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import com.palantir.atlasdb.keyvalue.api.Namespace;
+import com.palantir.atlasdb.timelock.auth.api.Client;
+import com.palantir.atlasdb.timelock.auth.api.NamespaceMatcher;
+
+public class PrivilegeBasedAuthRequirer implements AuthRequirer {
+    private List<NamespaceMatcher> authorizedNamespaces;
+
+    PrivilegeBasedAuthRequirer(Map<Client, NamespaceMatcher> privileges) {
+        this.authorizedNamespaces = new ArrayList<>(privileges.values());
+    }
+
+    @Override
+    public boolean requiresAuth(Namespace namespace) {
+        return authorizedNamespaces.stream().anyMatch(namespaceMatcher -> namespaceMatcher.matches(namespace));
```

Comment:
Maybe premature, but at some point this feels like `LoadingCache` material. It's a little unfortunate, but I don't see a good way to make the computation faster

Reply:
will add a cache here, but want to do it in a separate pr, I first need to clarify the live-reloading part.